### PR TITLE
convertTo: correctly handle uint8_t and int8_t

### PR DESCRIPTION
**Pull request by dawagner**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/280%23issuecomment-147698349%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/280%23discussion_r41879325%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/280%23discussion_r41879908%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/280%23issuecomment-147760628%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/280%23issuecomment-147967252%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/280%23issuecomment-147982910%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/280%23issuecomment-147698349%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40krocard%20%40tcahuzax%20please%20review%22%2C%20%22created_at%22%3A%20%222015-10-13T12%3A16%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40krocard%20wish%20granted%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-10-13T16%3A03%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-10-14T08%3A01%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-10-14T08%3A59%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2084512881b6b8c0046d1c40e84665c04602a56123%20utility/convert.hpp%2046%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/280%23discussion_r41879908%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22how%20about%20min%20bound%20%3F%22%2C%20%22created_at%22%3A%20%222015-10-13T15%3A16%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20utility/convert.hpp%3AL102-132%22%7D%2C%20%22Pull%2084512881b6b8c0046d1c40e84665c04602a56123%20utility/convert.hpp%2036%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/280%23discussion_r41879325%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22One%20day%20we%20should%20use%20%60static_assert%60.%22%2C%20%22created_at%22%3A%20%222015-10-13T15%3A12%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20utility/convert.hpp%3AL102-132%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 84512881b6b8c0046d1c40e84665c04602a56123 utility/convert.hpp 36'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/280#discussion_r41879325'>File: utility/convert.hpp:L102-132</a></b>
- <a href='https://github.com/krocard'></a> One day we should use `static_assert`.
- [ ] <a href='#crh-comment-Pull 84512881b6b8c0046d1c40e84665c04602a56123 utility/convert.hpp 46'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/280#discussion_r41879908'>File: utility/convert.hpp:L102-132</a></b>
- <a href='https://github.com/krocard'></a> how about min bound ?
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/280#issuecomment-147698349'>General Comment</a></b>
- <a href='https://github.com/dawagner'></a> @krocard @tcahuzax please review
- <a href='https://github.com/dawagner'></a> @krocard wish granted :)

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/280?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/280?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/280'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/280'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by dawagner** [comment]
@krocard @tcahuzax please review

File: utility/convert.hpp [hunk 1]
```diff
@@ -94,6 +102,30 @@ static inline bool convertTo(const std::string &str, T &result)
...
     return ss.eof() && !ss.fail() && !ss.bad();
 }
+
+template<typename T, typename Via>
+static inline bool convertToVia(const std::string &str, T &result)
+{
+    /* Check that conversion to that type is allowed.
+     * If this fails, this means that this template was not intended to be used
+     * with this type, thus that the result is undefined. */
+    ConvertionAllowedVia<T, Via>();
```
Comment (krocard) [inline utility/convert.hpp]: One day we should use `static_assert`.

File: utility/convert.hpp [hunk 2]
```diff
@@ -94,6 +102,30 @@ static inline bool convertTo(const std::string &str, T &result)
...
+
+    /* We want to override the behaviour of convertTo<T> with that of
+     * convertTo<Via> and then safely cast the result into a T. */
+    Via res;
+
+    if (!convertTo<Via>(str, res)) {
+        return false;
+    }
+
+    if (res > std::numeric_limits<T>::max()) {
```
Comment (krocard) [inline utility/convert.hpp]: how about min bound ?

**Comment by dawagner** [comment]
@krocard wish granted :)

**Comment by tcahuzax** [comment]
:+1:

**Comment by krocard** [comment]
:+1:

File: utility/convert.hpp (hunk 2 above)
Comment (krocard) [inline utility/convert.hpp]: :+1:<a href='#crh-update-none'></a>

File: utility/convert.hpp (hunk 1 above)
Comment (krocard) [inline utility/convert.hpp]: :+1:<a href='#crh-update-complete'></a>

**Pull request merged**